src/codec.rs

```diff
@@ -96,8 +98,12 @@ impl From<&'static str> for Error {
 
 /// Trait that allows reading of data into a slice.
 pub trait Input {
-	/// Return remaining length of input.
-	fn remaining_len(&mut self) -> Result<usize, Error>;
+	/// If input has a length then return the remaining length of input.
```

Comment:
```suggestion
	/// Should return the remaining length of the input data. If no information about the input length is available, `None` should be returned.
	///
	/// The length is used to constrain the preallocation while decoding. Returning a garbage length can open the doors for a denial of service attack to your application. 
	/// Otherwise, returning `None` can decrease the performance of your application.
```